src/lib/udev/google_nvme_id

```diff
@@ -0,0 +1,305 @@
+# Globals:
+#   Populates ID_NAMESPACE_INDEX, ID_SERIAL_SHORT, and ID_SERIAL
+# Arguments:
+#   The device path for the disk
+# Returns:
+#   0 on success and 255 if an error occurrs
+#######################################
+function identify_pd_disk() {
+  local dev_path="${1}"
+  local dev_name
+  dev_name="$(get_namespace_device_name "${dev_path}")"
+  if [[ $? != 0 ]]; then
+    return 255
+  fi
+
+  ID_NAMESPACE_INDEX="${namespace_index}"
+  ID_SERIAL_SHORT="${dev_name}"
+  ID_SERIAL="Google_PersistentDisk_${ID_SERIAL_SHORT}"
+  return 0
+}
+
+#######################################
+# Ensures a given command/path exists
+# Arguments:
+#   The path for the command
+# Returns:
+#   0 if the command is found/executable and 1 on error
+#######################################
+function confirm_command() {
+  local command_to_check="${1}"
+  command -v "${command_to_check}" > /dev/null 2>&1
+  if [[ $? != 0 ]]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+function print_help_message() {
+  echo "Usage: google_nvme_id [-s] [-h] -d device_path"
+  echo "  -d <device_path> (Required): Specifies the path to generate a name"
+  echo "        for.  This needs to be a path to an nvme device or namespace"
+  echo "  -s: Create symbolic link for the disk under /dev/disk/by-id."
+  echo "        Otherwise, the disk name will be printed to STDOUT"
+  echo "  -h: Print this help message"
+}
+
+function main() {
+  local opt_gen_symlink='false'
+  local device_path=''
+
+  while getopts :d:sh flag; do
+    case "${flag}" in
+      d) device_path="${OPTARG}";;
+      s) opt_gen_symlink='true';;
+      h) print_help_message
+         return 0
+         ;;
+      :) echo "Invalid option: ${OPTARG} requires an argument" 1>&2
+         return 1
+         ;;
+      *) return 1
+    esac
+  done
+
+  if [[ -z "$device_path" ]]; then
+    echo "Device path (-d) argument required. Use -h for full usage." 1>&2
+    exit 1
+  fi
+
+  confirm_command "${nvme_cli_bin}"
+  if [[ $? != 0 ]]; then
+    local err_msg="The nvme utility (/usr/sbin/nvme) was not found."
+    err_msg="${err_msg} You may need to run with sudo or install nvme-cli."
+    err "${err_msg}"
+    return 1
+  fi
+
+  # Ensure the passed device is actually an NVMe device
+  validate_nvme_device "${device_path}"
+  if [[ $? != 0 ]]; then
+    local err_msg="Passed device was not an NVMe device.  (You may need to run"
+    err_msg="${err_msg} this script as root/with sudo)."
+    err "${err_msg}"
+    return 1
+  fi
+
+  # Detect the type of attached nvme device
+  local device_type="$(detect_pd_or_local_ssd "${device_path}")"
+  local device_type_ret_value=$?
+  if [[ "${device_type}" == "${LOCAL_SSD}" ]]; then
+    err "This script only supports PD NVMe disks but given path was Local SSD"
+    return 1
+  elif [[ "${device_type_ret_value}" != 0 ]]; then
+    err "Unknown device type (neither a PD nor Local SSD NVMe device)"
+    return 1
+  fi
```

Comment:
this is overcomplicated, imo. the detect_pd_or_local_ssd function checks for two hardcoded values, translating them into one of two different globals. then in use here, the two globals are checked against. so the globals were not used, and we could have just checked against the real command output here directly

try:
```
local device_type
if ! device_type=$("$nvme_cli_bin" id-ctrl "$device_path"); then
  err "Unknown device type [...]"
  return 1
fi
if [[ "$device_type" != "nvme_card-pd" ]]; then
  err "Unsupported device type [...]"
  return 1
fi
```

Reply:
Updated to remove the globals/detect_pd_or_local_ssd function.  These globals were originally used in multiple places (when the script was also used for local SSD).